tests/ttnn/unit_tests/test_matmul.py

```diff
@@ -287,3 +287,38 @@ def test_tutorial_matmul_with_tilized_input_in_l1_memory_and_user_specified_core
     output = ttnn.to_torch(output)
 
     assert_with_pcc(torch_output_tensor, output, pcc=0.999)
+
+
+@skip_for_wormhole_b0()
+def test_sharded_matmul(device):
+    torch.manual_seed(0)
+    grid_size = (5, 8)
+
+    M = 1600
+    K = 256
+    N = 1024
+    in0_shape = [1, 1, M, K]
+    in1_shape = [1, 1, K, N]
+
+    torch_input_tensor_a = torch.randn(in0_shape, dtype=torch.bfloat16)
+    torch_input_tensor_b = torch.randn(in1_shape, dtype=torch.bfloat16)
+    torch_output_tensor = torch_input_tensor_a @ torch_input_tensor_b
+
+    input_tensor_a = ttnn.from_torch(torch_input_tensor_a)
+    input_tensor_b = ttnn.from_torch(torch_input_tensor_b)
+
+    input_tensor_a = ttnn.to_device(input_tensor_a, device, memory_config=ttnn.L1_MEMORY_CONFIG)
+    input_tensor_b = ttnn.to_device(input_tensor_b, device, memory_config=ttnn.L1_MEMORY_CONFIG)
+
+    input_tensor_a = ttnn.to_layout(input_tensor_a, ttnn.TILE_LAYOUT)
+    input_tensor_b = ttnn.to_layout(input_tensor_b, ttnn.TILE_LAYOUT)
+
+    sharded_mem_config = ttnn.BLOCK_SHARDED_MEMORY_CONFIG(grid_size, [M // grid_size[0], K // grid_size[1]])
```

Comment:
Does `[M // grid_size[0], K // grid_size[1]]` needed as an extra argument? Can we ever pass in any other values?